src/data/content-items/data-source.js

```diff
@@ -542,4 +542,45 @@ export default class ContentItem extends oldContentItem.dataSource {
     );
     return notes;
   };
+
+  // Had to override this function to be able to remove campaign channel siblings
+  getCursorBySiblingContentItemId = async (id) => {
```

Comment:
what part of this function is custom? I'd really like to investigate if we can avoid copy and pasting this thing, it's massive and will inevitably lead to missed updates down the road.

Reply:
Yep, the custom part is the `blacklistedIds`, `finalIds`, and using `finalIds` in the return `map`. Vincent said they did something similar in one of the other apps he'd been working on to do the same thing. But if it should live in core then all the better.

Reply:
Couldn't you prevent campaign children from getting in by overriding the `siblingConnection` like you did with the `childrenConnection`?

Reply:
I don't think so? The `childContentItemsConnection` just sorted the data after the `getCursorByParentContentItemId` function actually got the data. This function is the function that gets the data for the `siblingContentItemsConnection`. Doesn't make a lot of sense to me to get all the data from this function in core and then have to go through it all a second time in the resolver function to take out the pieces that don't belong.

Reply:
Now granted, it would be a lot smaller bit of code if we could pull it off.

Reply:
in the same way that you altered the cursor before you fetched, couldn't you just an extra `filter` that excludes any content with the campaign content channel? 

said another way, for the children connection, you reversed the sort order, for the sibling can't you just exclude any content from the campaign channel?

Reply:
`cursor.filter(ContentChannelId ne ${CAMPAIGN})`

Reply:
Taking a look at this, I don't think I can filter _after_ the cursor has been set. The `getCursorBySiblingContentItemId` function does all the necessary filtering (getting the items from the content channels it needs to) and returns the items without information about the parent channel it came from. 

Here's the data that's returned by that function: https://gist.github.com/richarddubay/a4c686d8be5e07ab1077c567f6af201b

There's really nothing there to `filter` on after the fact.

Reply:
just like you are adding an extra sort to this cursor, can you not add a filter for content channel ID to the sibling cursor?

<img width="604" alt="Screen Shot 2020-06-08 at 11 52 06 AM" src="https://user-images.githubusercontent.com/2659478/84052528-b2dca700-a97e-11ea-8837-f0e615988e74.png">

Reply:
![Screenshot 2020-06-08 13 22 07](https://user-images.githubusercontent.com/3229463/84060965-12d94a80-a98b-11ea-83e5-0602e576085f.png)

I get what you're asking but the `contentChannelId` here is the `id` of the sermons content channel (19). `root.contentChannelId` is 19. The `contentChannelId` in each of the items in the returned cursor data is 19. As far as I can tell the only place that I have access to the campaign channel id is in the other function.

Reply:
campaign channel id is hardcoded in the config yaml so you just filter any content that has that as its channel

Reply:
or you could filter on matching content channels so if the root is 19, you only allow content with channel 19 through

Reply:
That's what I'm saying though. Look at the data in the gist that I posted earlier. There are 3 items in there, all with `contentChannelId` of 19. All 3 of those items are in the campaign channel but only 2 of them are actually related and should show together. All the filtering is happening on the `getCursorBySiblingContentItemId` function and there is nothing being sent down (root or otherwise) that contains the campaign channel id once the cursor is created. There's literally nothing to filter on once the data is returned from that function. 

Maybe I'm misunderstanding what you're saying but I don't think I am. I just don't have anything that I can filter on that would remove the items that don't belong from the cursor. You're trying to get rid of a large function override ... which I get. But I don't know how to make it happen.